src/module/applications/api/document-sheet.mjs

```diff
@@ -0,0 +1,527 @@
+    this.window.controls.after(...buttons);
+
+    return frame;
+  }
+
+  /* -------------------------------------------------- */
+
+  /** @inheritdoc */
+  async _prepareContext(options) {
+    const context = await super._prepareContext(options);
+    Object.assign(context, {
+      isPlay: this.isPlayMode,
+      owner: this.document.isOwner,
+      limited: this.document.limited,
+      gm: game.user.isGM,
+      document: this.document,
+      system: this.document.system,
+      systemSource: this.document.system._source,
+      systemFields: this.document.system.schema.fields,
+      flags: this.document.flags,
+      config: ds.CONFIG,
+    });
+    return context;
+  }
+
+  /* -------------------------------------------------- */
+  /*   Actions                                          */
+  /* -------------------------------------------------- */
+
+  /**
+   * Toggle Edit vs. Play mode.
+   *
+   * @this DSDocumentSheet
+   * @param {PointerEvent} event   The originating click event.
+   * @param {HTMLElement} target   The capturing HTML element which defined a [data-action].
+   */
+  static async #toggleMode(event, target) {
+    if (!this.isEditable) {
+      console.error("You can't switch to Edit mode if the sheet is uneditable");
+      return;
+    }
+    await this.render({ mode: this.isPlayMode ? DSDocumentSheet.MODES.EDIT : DSDocumentSheet.MODES.PLAY });
+  }
+
+  /* -------------------------------------------------- */
+
+  /**
+   * Renders an embedded document's sheet in play or edit mode based on the document sheet view mode.
+   *
+   * @this DSDocumentSheet
+   * @param {PointerEvent} event   The originating click event.
+   * @param {HTMLElement} target   The capturing HTML element which defined a [data-action].
+   * @protected
+   */
+  static async #viewDoc(event, target) {
+    const doc = this._getEmbeddedDocument(target);
+    if (!doc) {
+      console.error("Could not find document");
+      return;
+    }
+    await doc.sheet.render({ force: true, mode: this._mode });
+  }
+
+  /* -------------------------------------------------- */
+
+  /**
+   * Handles document deletion.
+   *
+   * @this DSDocumentSheet
+   * @param {PointerEvent} event   The originating click event.
+   * @param {HTMLElement} target   The capturing HTML element which defined a [data-action].
+   * @protected
+   */
+  static async #deleteDoc(event, target) {
+    const doc = this._getEmbeddedDocument(target);
+    if (doc.hasGrantedItems) await doc.advancementDeletionPrompt();
+    else await doc.deleteDialog();
+  }
+
+  /* -------------------------------------------------- */
+
+  /**
+   * Handle creating a new embedded document using initial data defined in the HTML dataset.
+   *
+   * @this DSDocumentSheet
+   * @param {PointerEvent} event   The originating click event.
+   * @param {HTMLElement} target   The capturing HTML element which defined a [data-action].
+   * @private
+   */
+  static async #createDoc(event, target) {
```

Comment:
Confused as to whats up in this method, it references both `this.item` and `this.actor`.